### PR TITLE
Add automatic layout creation when creating Java activity class. [Fixed]

### PR DESCRIPTION
**Pull request by MatheusTGamerPro**
this PR adds the layout creation along with the Java activity one (The past PR ended up having conflicts that can't be fixed, that's why I'm doing this one)

Refers to the issue: #244

### CODE REVIEW

File: app/src/main/java/com/itsaky/androidide/handlers/FileTreeActionHandler.kt [hunk 1]
```diff
@@ -336,6 +356,7 @@ class FileTreeActionHandler : BaseEventHandler() {
       if (newFile.exists()) {
         app.toast(string.msg_file_exists, ERROR)
       } else {
+        if (autoLayout) createAutoLayout(context, directory, name)
```
Comment (itsaky) [inline app/src/main/java/com/itsaky/androidide/handlers/FileTreeActionHandler.kt:359]: If the layout file creation fails for any reason (let's say the layout file already exists), then the process should probably fail. Also, if the layout file exists, you show an error message to the user, but still the Java file creation process continues. This shouldn't happen.

If the user has selected to create the layout file, and the layout file exists, the process should simply fail.

File: app/src/main/java/com/itsaky/androidide/handlers/FileTreeActionHandler.kt [hunk 2]
```diff
@@ -253,6 +257,22 @@ class FileTreeActionHandler : BaseEventHandler() {
...
+    val newFileLayout = File(projectDir, layoutName)
+    if (newFileLayout.exists()) {
+      app.toast(string.msg_file_exists, ERROR)
+    } else {
+      if (FileIOUtils.writeFileFromString(newFileLayout, ProjectWriter.createLayout())) {
+        notifyFileCreated(newFileLayout, context)
+      } else {
+        app.toast(string.msg_file_creation_failed, ERROR)
+      }
+    }
```
Comment (itsaky) [inline app/src/main/java/com/itsaky/androidide/handlers/FileTreeActionHandler.kt:273]: Use `if-return` statements for better readability. For example :
```java
if (!newFileLayout.exists()) {
  // show error
  return;
}

if (!writeFile(...)) {
  // show error
  return;
}

// notify file created
```

File: app/src/main/res/layout/layout_create_file_java.xml [hunk 3]
```diff
@@ -83,5 +83,15 @@
 
   </com.google.android.material.textfield.TextInputLayout>
 
+  <CheckBox
+        android:id="@+id/checkButton"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="16dp"
+        android:layout_marginStart="12dp"
```
Comment (itsaky) [inline app/src/main/res/layout/layout_create_file_java.xml:90]: Why use `16dp` for end margin but `12dp` for start?

**Review by itsaky — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -184,6 +184,74 @@
...
+  <string name="classtype_class">Class</string>
+  <string name="classtype_activity">Activity</string>
+  <string name="classtype_interface">Interface</string>
+  <string name="classtype_enum">Enum</string>
+  <string name="msg_checking_storage_permissions">Checking storage permissions</string>
+  <string name="msg_failed_list_files">Failed to list project files!</string>
+  <string name="msg_storage_denied">Can\'t proceed without storage permissions</string>
+  <string name="msg_storage_granted">Storage permissions granted!</string>
+
   <!-- Code Editor -->
```
Comment (itsaky) [inline app/src/main/res/values/strings.xml:255]: Why is this duplicated?

File: app/src/main/res/layout/layout_create_file_java.xml (hunk 3 above)
Comment (MatheusTGamerPro) [reply app/src/main/res/layout/layout_create_file_java.xml:90]: Which would you like best for the layout with the new material design?

**Review by itsaky — APPROVED** [review]

**Pull request merged**